CONTRIBUTING.md

```diff
@@ -3,180 +3,84 @@
-Before you start working on it, make sure that it's not already
-assigned to someone and that nobody has left a comment saying that
-they are working on it!  (Of course, you can also comment on an issue
-someone is already working on and offer to collaborate.)
-
-Have an idea for something new? That's great! We recommend that you
-make sure it belongs in http4s before you put effort into creating a
-pull request. The preferred ways to do that are to either:
-
- * [create a GitHub issue] describing your idea.
- * get feedback in the [http4s Gitter room](https://gitter.im/http4s/http4s).
-
-### Let us know you are working on it
+### Claim an issue
 
 If there is already a GitHub issue for the task you are working on,
-leave a comment to let people know that you are working on it. If
-there isn't already an issue and it is a non-trivial task, it's a good
-idea to create one (and note that you're working on it). This prevents
-contributors from duplicating effort.
-
-### Build the project
+please leave a comment to let people know that you are working on it.
+If there isn't already an issue and it is a non-trivial task, it's a
+good idea to create one (and note that you're working on it). This
+prevents contributors from duplicating effort.
 
-First you'll need to checkout a local copy of the code base:
+### Targeting a branch
 
-```sh
-git clone git@github.com:http4s/http4s.git
-```
+http4s actively maintains three branches:
 
-To build http4s you should have [sbt installed].  Run `sbt`, and then
-use any of the following commands:
+* [release-0.15.x]: the current production release.  Merges to [master].
+* [master]: the current development mainline.  Merges to [cats].
+* [cats]: the [cats-fs2] port
 
- * `compile`: compile the code
- * `console`: launch a REPL
- * `test`: run the tests
- * `makeSite`: generate the documentation, including scaladoc.
- * `scalastyle`: run the style-checker on the code
+The guide below helps find the most appropriate branch for your change.
 
-### Write code
+My change is...                               | Branch
+----------------------------------------------|-------------------
+Documentation of existing features            | [release-0.15.x]
+Documentation of unreleased features          | [master]
+Change to docs/src/site                       | [master]
+Specific to cats or fs2                       | [cats]
+Binary compatible with current release        | [release-0.15.x]
+Binary incompatible with current release      | [master]
 
-TODO
+Still unsure?  Don't worry!  Send us that PR, and we'll cherry-pick it
+to the right place.
 
 ### Attributions
 
 If your contribution has been derived from or inspired by other work,
-please state this in its ScalaDoc comment and provide proper
+please state this in its scaladoc comment and provide proper
 attribution. When possible, include the original authors' names and a
 link to the original work.
 
-### Write tests
```

Comment:
I'd leave this part in, a few basic pointers to where to start testing are always helpful.